internal/lang/marks/marks.go

```diff
@@ -110,7 +116,11 @@ func DeprecatedOutput(v cty.Value, addr addrs.AbsOutputValue, msg string, isFrom
 	return Deprecated(v, DeprecationCause{
 		IsFromRemoteModule: isFromRemoteModule,
 		By:                 callOutAddr,
-		Message:            msg,
+		// Used to identify the output on the consolidation diagnostics and
+		// make sure they are consolidated correctly. We use:
+		// output value + deprecated message as the key
+		Key:     addr.OutputValue.Name + msg,
```

Comment:
I'd suggest including a newline character, or really just any other character that isn't allowed in an HCL identifier, to separate these two parts.

Although I acknowledge that it's very unlikely, without some sort of delimiter it's possible that two combinations of different output value name and message would result in the same concatenated string and so be incorrectly consolidated. That would not cause a huge problem but might nonetheless be confusing and hard to debug.